libmamba/src/api/env.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include <iostream>
+
+#include "mamba/api/configuration.hpp"
+#include "mamba/api/env.hpp"
+#include "mamba/core/environments_manager.hpp"
+#include "mamba/util/string.hpp"
+
+namespace mamba
+{
+    void print_envs(Configuration& config)
+    {
+        config.load();
+        mamba::detail::print_envs_impl(config);
+    }
+
+    namespace detail
+    {
+        std::string get_env_name(const Context& ctx, const mamba::fs::u8path& px)
+        {
+            auto& ed = ctx.envs_dirs[0];
+            if (px == ctx.prefix_params.root_prefix)
+            {
+                return "base";
+            }
+            else if (util::starts_with(px.string(), ed.string()))
+            {
+                return mamba::fs::relative(px, ed).string();
+            }
+            else
+            {
+                return "";
+            }
+        }
+
+        void print_envs_impl(const Configuration& config)
```

Comment:
Sorry, I think I should have been clearer: I meant that the signature should be changed, but other changes for the signature of this definition are necessary (_e.g._ the signature of the declaration of `print_envs_impl` at least must be changed).

(Also it seems that a `const-qualified l-value reference to the instance of `Context` could be passed.)